universe/supplycommit/supply_commit_transitions.go

```diff
@@ -0,0 +1,804 @@
+	nodeSum := rootNode.NodeSum()
+
+	// Count the different types of updates included in this commitment.
+	var mints, burns, ignores int
+	for _, update := range supplyTransition.PendingUpdates {
+		switch update.SupplySubTreeType() {
+		case MintTreeType:
+			mints++
+		case BurnTreeType:
+			burns++
+		case IgnoreTreeType:
+			ignores++
+		}
+	}
+
+	// Format the label string including the asset specifier, root hash, and
+	// counts of each update type.
+	label := fmt.Sprintf(
+		"supply_commit(%v):root=%x:sum=%v:m=%d,b=%d,i=%d",
+		assetSpec, rootHash[:], nodeSum, mints, burns, ignores,
+	)
+
+	return label
+}
+
+// ProcessEvent is used to transition from the default state to the
+// UpdatePendingState once we receive a new request to update the current supply
+// trees.
+func (d *DefaultState) ProcessEvent(event Event,
+	env *Environment) (*StateTransition, error) {
+
+	// In this state, we'll receive one of three events: a new burn, a new
+	// mint, or a new ignore. For all three types, we'll emit this as an
+	// internal event as we transition to the UpdatePendingState.
+	switch supplyEvent := event.(type) {
+	case SupplyUpdateEvent:
+
+		// Before we transition to the next state, we'll add this event
+		// to our update log. This ensures that we'll remember to
+		// process from this state after a restart.
+		//
+		// TODO(roasbeef): special error case here?
+		ctx := context.Background()
+		err := env.StateLog.InsertPendingUpdate(
+			ctx, env.AssetSpec, supplyEvent,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to insert "+
+				"pending update: %w", err)
+		}
+
+		// We'll go to the next state, caching the pendingUpdate we just
+		// received.
+		return &StateTransition{
+			NextState: &UpdatesPendingState{
+				pendingUpdates: []SupplyUpdateEvent{
+					supplyEvent,
+				},
+			},
+		}, nil
+
+	// If we get a tick in this state, then it's just a no-op. We'll
+	// transition back to the same state.
+	case *CommitTickEvent:
+		return &StateTransition{
+			NextState: d,
+		}, nil
+
+	// Any other messages in this state will result in an error, as this is
+	// an undefined state transition.
+	default:
+		return nil, fmt.Errorf("%w: received %T while in %T",
+			ErrInvalidStateTransition, event, d)
+	}
+}
+
+// ProcessEvent processes incoming events for the UpdatesPendingState. From
+// here, we'll either be staging new events, or we'll attempt to start the
+// commitment update process once a new commitment is received.
+func (s *UpdatesPendingState) ProcessEvent(event Event, env *Environment) (
+	*StateTransition, error) {
+
+	switch newEvent := event.(type) {
+	// We've received a new update event, we'll stage this in our local
+	// state, and do a self transition.
+	case SupplyUpdateEvent:
+
+		// We just got a new pending update in addition to the one that
+		// made us transition to this state. This ensures that we'll
+		// remember to process from this state after a restart.
+		ctx := context.Background()
```

Comment:
I feel like we should make the context part of the `ProcessEvent` argument list, so we can pass it in instead of needing to create a new one every time.

Reply:
I think there's actually a pending PR open on `lnd` to do this. Agree it makes sense to pass in generically.